File: data-structures-and-algorithms-ashrafi/java/src/recursive/problem7/Problem7.java
package recursive.problem7;

public class Problem7 {
    
    public static int gcd(int a, int b){
        if (b == 0) {
            return a;
        }
        return gcd(b, a % b);
    }

    public static void main(String[] args) {
        int num1 = 18;
        int num2 = 12;

        System.out.printf("(%d, %d) = %d", num1, num2, gcd(num1, num2));
    }
}

File: data-structures-and-algorithms-ashrafi/python/src/recursive/problem_7/problem_7.py
def gcd(a, b):
    if b == 0:
        return a
    return gcd(b, a % b)


def main():
    num_1 = 18
    num_2 = 12

    print(f"({num_1}, {num_2}) = {gcd(num_1, num_2)}")
    
    
if __name__ == "__main__":
    main()